Feature/refactor lib

Hellyson-Ferreira opened the pull request:
Added: 
     - Method for list all files of a video
     - Method upload_poster
     - Method get_url_posters
     - Method upload video by url
     -  Class VideoBandwidth
     - Method bandwidth
Updated:
     - RAEDEME
     - method  delete_tag
     - method  delete_all_tags

Hellyson-Ferreira committed:
added method delete_subtitle

Hellyson-Ferreira committed:
Added method add_tag

Hellyson-Ferreira committed:
Added:
   - Method for search with tag

Hellyson-Ferreira committed:
Added:
   - Method for list of all tags

Hellyson-Ferreira committed:
Replace tag

Commit:
fix.

Hellyson-Ferreira committed:
Update:
   - README
Added:
   - Method delete_all

Hellyson-Ferreira committed:
Update:
   - README
   - conversation resolved

Hellyson-Ferreira committed:
Update:
   - README
   - tests method
   - class Video

Hellyson-Ferreira committed:
Merge with origin/develop

Hellyson-Ferreira committed:
Added:
  - Method for list all files of a video

updated:
   - RAEDEME
   - Rename method

Hellyson-Ferreira committed:
Added:
  - Method upload_poster
  - Method get_url_posters

updated:
   - RAEDEME
   - method  delete_tag
   - method  delete_all_tags

Hellyson-Ferreira committed:
Added:
  - Method upload by url
updated:
   - RAEDEME

Hellyson-Ferreira committed:
Added:
  - Method bandwidth

Hellyson-Ferreira committed:
Added:
  - Class VideoBandwidth
  - test for class VideoBandwidth
Updated:
  -Method bandwidth

Hellyson-Ferreira committed:
Updated:
  -Method bandwidth added type

Hellyson-Ferreira committed:
Updated:
  - README

fariias commented on vdocipher/resources/bandwith.py:
```diff
@@ -0,0 +1,35 @@
+@dataclass
+class VideoBandwidth:
+    data: str = None
+    video_id: str = None
```
remover o video_id e o video_titlte desse model.

fariias commented on vdocipher/resources/video.py:
```diff
@@ -61,6 +85,13 @@ class Video:
+    ip_geo_rules: List[IPGeoRule] = field(metadata=config(field_name="ipGeoRules"), default=None)
+    posters: List = None
+    poster: str = None
+    total_sinze_bytes: int = field(metadata=config(field_name="totalSinzeBytes"), default=None)
```
size**

fariias commented on vdocipher/resources/video.py:
```diff
@@ -144,7 +201,7 @@ def delete_tag(self):
         return response.json()
 
-    def delete_tag_to_video_ids(self, videos_id: List = None):
+    def delete_all_tag_to_video_ids(self, videos_id: List = None):
```
by_video_Ids**

fariias commented on vdocipher/resources/bandwith.py:
```diff
@@ -0,0 +1,35 @@
+    video_id: str = None
+    video_title: str = None
+    bandwidth: str = None
+
```
Adicionar property video() que faz o get e retorna o video correspondente ao bandwith

fariias commented on vdocipher/resources/video.py:
```diff
@@ -221,8 +286,40 @@ def delete(self):
         )
         return response
 
+    def list_all_files(self) -> List[VideoFiles]:
```
talvez mudar para get...

fariias commented on vdocipher/resources/video.py:
```diff
@@ -221,8 +286,40 @@ def delete(self):
+
+        return response.json()['posters']
+
+    def bandwidth(self, date_filter: date = None) -> VideoBandwidth:
```
renomear para get_bandwith e adicionar bandwith como propriedade da classe.

fariias commented on vdocipher/tests/test_video.py:
```diff
@@ -23,6 +27,13 @@ def test_video_get_list_pagination(self):
 
         [video.delete() for video in video_list]
 
+    def test_video_get_all_video(self, video: Video):
```
videos**

fariias replied:
adicionar mais videos ex: 4 e ver se o resultado traz todos

fariias commented on vdocipher/tests/test_video.py:
```diff
@@ -41,6 +52,14 @@ def test_video_upload(self, video: Video):
 
+    def test_upload_by_url(self):
+        response = self.vdocipher.Video().upload_by_url(
+            'https://instagram.fthe11-1.fna.fbcdn.net/v/t50.2886-16/179158289_4317944001571515_5690961565767437429_n.mp4?_nc_ht=instagram.fthe11-1.fna.fbcdn.net&_nc_cat=103&_nc_ohc=dH-FDZ1QSJ8AX8oukVC&edm=AP_V10EBAAAA&ccb=7-4&oe=60A0B2FC&oh=a44759fa9467621b57c09af7dc2fa703&_nc_sid=4f375e')
```
talvez mudar o link para algo mais seguro, ex: um video do youtube.

fariias commented on vdocipher/tests/test_video.py:
```diff
@@ -123,12 +153,16 @@ def test_get_all_tags(self):
     def test_repalce_tag(self, video):
```
replace**

fariias commented on vdocipher/tests/test_video.py:
```diff
@@ -142,32 +176,95 @@ def test_replace_tag_to_video_ids(self):
         assert otp.otp
         assert otp.playback_info
+
+    def test_list_all_files(self, video):
```
adicionar mais arquivos para testar, e ver se eles vem.

fariias commented on vdocipher/tests/test_video.py:
```diff
@@ -142,32 +176,95 @@ def test_replace_tag_to_video_ids(self):
+        response = video.list_all_files()
+        assert len(response) > 0
+
+    def test_upload_post(self, video: Video):
```
upload_poster**

Hellyson-Ferreira committed:
Added:
  - Method delete_tag_to_video_ids
  - property video in class VideoBandwidth

Updated:
  - README

Hellyson-Ferreira committed:
- handling exceptions

Hellyson-Ferreira committed:
- handling exceptions

Hellyson-Ferreira committed:
- test poster image in format jpg

fariias commented on README.md:
```diff
@@ -117,6 +138,20 @@ otp = Video(title='test video').upload('file').create_otp()
  otp = vdocipher.OTP(ip_geo_rule=geo_rules_list).create(videoid='your_video_id')
  
+ # obtaining video bandwidth
+ video_obj = vdocipher.Video().get()
```
Acho que faltou o id do video

fariias commented on README.md:
```diff
@@ -66,12 +72,26 @@ tag_list_replace = ['Python', 'Rust', 'TypeScript']
 
-# deleteingg tag in multiple videos
+# deleting a tags in multiple videos
+vdocipher.Video().delete_tag_by_video_ids(videos_id=video_list_id, tag='PythonJS')
```
Talvez colocar uma lista de ids de exemplos

fariias commented on vdocipher/resources/bandwith.py:
```diff
@@ -0,0 +1,43 @@
+
+
+@dataclass
+class VideoBandwidth:
```
N seria VideoBandwith? Huhuhuhu

fariias commented on vdocipher/resources/video.py:
```diff
@@ -144,7 +201,19 @@ def delete_tag(self):
         return response.json()
 
-    def delete_tag_to_video_ids(self, videos_id: List = None):
+    def delete_tag_by_video_ids(self, videos_id: List = None, tag: str = None):
```
Acha que fica mais semântico se colocarmos video_ids?

fariias commented on vdocipher/tests/test_bandwidth.py:
```diff
@@ -0,0 +1,35 @@
+from datetime import date
+
+
+class TestVideoBandwidth(BaseTest):
```
Bandwidth

fariias commented on vdocipher/tests/test_bandwidth.py:
```diff
@@ -0,0 +1,35 @@
+
+class TestVideoBandwidth(BaseTest):
+
+    def test_get(self):
```
Vamos ter que fazer um mock nesse cara pra n ficar com o teste assimcom a data na mão, vou te mostrar como

Hellyson-Ferreira committed:
Resolved conversation

Hellyson-Ferreira committed:
Resolved conversation

Hellyson-Ferreira committed:
Resolved tests
